test/helpers/kubectl.go

```diff
@@ -934,83 +933,83 @@ func (kub *Kubectl) ciliumInstall(dsPatchName, cmPatchName string, getK8sDescrip
+	}
+
 	return nil
 }
 
-// CiliumOperatorInstall installs Cilium operator if it is supported/required
-// by versionTag.
-func (kub *Kubectl) CiliumOperatorInstall(versionTag string) (installed bool, err error) {
-	var (
-		ciliumOperatorSampleFile   string
-		ciliumOperatorPatchFile    string
-		ciliumOperatorSaSampleFile string
+func addIfNotOverwritten(options []string, field, value string) []string {
+	for _, s := range options {
+		if strings.HasPrefix(s, "--set "+field) {
+			return options
+		}
+	}
 
-		ciliumOperatorYamlName   = "cilium-operator.yaml"
-		ciliumOperatorSAYamlName = "cilium-operator-sa.yaml"
-	)
+	options = append(options, "--set "+field+"="+value)
+	return options
+}
 
-	getFileFromOlderVersion := func(filename string) string {
-		return fmt.Sprintf("https://raw.githubusercontent.com/cilium/cilium/%s/examples/kubernetes/%s/%s",
-			versionTag, GetCurrentK8SEnv(), filename)
+// ciliumInstallHelm installs Cilium with the Helm options provided.  Returns an
+// error if any patch or if any original descriptors files were not found.
+func (kub *Kubectl) ciliumInstallHelm(options []string) error {
+	defaultOptions := map[string]string{
+		"global.hub":                    "k8s1:5000/cilium",
+		"agent.image":                   "cilium-dev",
+		"agent.tag":                     "latest",
+		"operator.image":                "operator",
+		"operator.tag":                  "latest",
+		"managed-etcd.hub":              "docker.io/cilium",
+		"global.debug":                  "true",
+		"global.k8s.requireIPv4PodCIDR": "true",
+		"global.pprof.enabled":          "true",
+		"global.logSystemLoad":          "true",
+		"global.bpf.preallocateMaps":    "true",
+		"global.etcd.leaseTTL":          "30s",
+		"global.ipv4.enabled":           "true",
+		"global.ipv6.enabled":           "true",
 	}
 
-	cst, _ := go_version.NewVersion("0")
+	for key, value := range defaultOptions {
+		options = addIfNotOverwritten(options, key, value)
+	}
 
-	if versionTag != "head" {
-		cst, err = go_version.NewVersion(versionTag)
-		if err != nil {
-			return false, fmt.Errorf("Not a valid version: %s", err)
-		}
-	}
-
-	switch {
-	case CiliumV1_0.Check(cst):
-		return false, nil
-	case CiliumV1_1.Check(cst):
-		return false, nil
-	case CiliumV1_2.Check(cst):
-		return false, nil
-	case CiliumV1_3.Check(cst):
-		return false, nil
-	case CiliumV1_4.Check(cst):
-		ciliumOperatorSampleFile = getFileFromOlderVersion(ciliumOperatorYamlName)
-		ciliumOperatorPatchFile = ""
-		ciliumOperatorSaSampleFile = getFileFromOlderVersion(ciliumOperatorSAYamlName)
-	default:
-		ciliumOperatorSampleFile = GetK8sDescriptor(ciliumOperatorYamlName)
-		ciliumOperatorPatchFile = ManifestGet("cilium-operator-patch.yaml")
-		ciliumOperatorSaSampleFile = GetK8sDescriptor(ciliumOperatorSAYamlName)
+	res := kub.ExecMiddle(fmt.Sprintf("helm template go/src/github.com/cilium/cilium/install/kubernetes/helm --namespace=kube-system %s > cilium.yaml", strings.Join(options, " ")))
```

Comment:
In the future we might want to use helm as a library and render it in code instead of running exec, I think it's beyond scope of this PR though. https://github.com/helm/helm/blob/master/pkg/renderutil/render.go

Can you add TODO comment here?

Reply:
That would be awesome! This is in line with our goal to remove the need to shell out and instead use APIs directly in the CI. Should I add this to the CI refactoring issue?

Reply:
Good point, added and filed https://github.com/cilium/cilium/pull/8637